### PR TITLE
Mastodon compat

### PR DESCRIPTION
**Pull request by Nutomic**
And it still doesnt manage to receive activities from Lemmy. Just returns status 202 with no error.

### CODE REVIEW

File: crates/apub_lib/src/signatures.rs [hunk 1]
```diff
@@ -46,7 +46,7 @@ pub async fn sign_and_send(
   headers.insert(HeaderName::from_str("Host")?, HeaderValue::from_str(&host)?);
   headers.insert(
     HeaderName::from_str("Date")?,
-    HeaderValue::from_str(&Utc::now().to_rfc2822())?,
+    HeaderValue::from_str(&Utc::now().to_rfc2822().replace("+0000", "GMT"))?,
```
Comment (Nutomic) [inline crates/apub_lib/src/signatures.rs:49]: Need to do this properly.

**Comment by Nutomic** [comment]
This is ready to merge, unless @asonix figures out how to get announces to work with masto.

**Comment by Nutomic** [comment]
Ready to merge and release. Just needs some more testing. Dont squash.

**Comment by Nutomic** [comment]
Announces arent shown properly on Pleroma, maybe we can also get that fixed.

File: crates/apub_lib/src/signatures.rs [hunk 2]
```diff
@@ -44,10 +44,9 @@ pub async fn sign_and_send(
     HeaderValue::from_str(APUB_JSON_CONTENT_TYPE)?,
   );
   headers.insert(HeaderName::from_str("Host")?, HeaderValue::from_str(&host)?);
-  headers.insert(
-    HeaderName::from_str("Date")?,
-    HeaderValue::from_str(&Utc::now().to_rfc2822())?,
-  );
+  // Need to use legacy timezone because mastodon and doesnt understand any new standards
```
Comment (dessalines) [inline crates/apub_lib/src/signatures.rs:47]: :(

**Review by dessalines — APPROVED** [review]

**Comment by Nutomic** [comment]
Announces on Pleroma seem to be working more or less, so we can release it as is, and maybe fix stuff later.

**Pull request merged**